### ISSUE TITLE
IEEE 754 not fully recognized

### ISSUE THREAD

**Issue by AlainBusser** [numworks/epsilon#320]
ln(0) yields undef whereas it should be -inf

1/inf yields undef whereas it should yield 0

exp(100000) yields undef whereas it should yield inf

inf+2 yields undef whereas it should yield inf

See page 17-18 of [this](http://perso.ens-lyon.fr/jean-michel.muller/goldberg.pdf) for more informations

**Comment by jacobly0** [comment]
I disagree with strictly following the IEEE 754 standard on a high school calculator since it can vary significantly from what is done in mathematics, and especially since it includes a programming language that already does.  That being said, these things in particular may make sense mathematically and should then be fixed for that reason, but not simply because IEEE 754 does it that way.  See https://github.com/numworks/epsilon/pull/119#issuecomment-330857565 for related discussion.

**Comment by AlainBusser** [comment]
Python does not quite follow the standard, which states that 1/0 should raise a division by zero exception, but return an infinity: Python only raises the exception and returns nothing. Idem for log(0) which should return -Infinity whereas Python only raises an exception (on the Numworks, because on [this online version](http://irem.univ-reunion.fr/blockly/plurialgo/blockly/extensions/sofuspy/run.html) I still have -inf as a result).

**Comment by AlainBusser** [comment]
@jacobly0 would you mind explaining more precisely how the standard is far from math teaching and what are the main difeerences between the two? I propose that you go to the end of [this page](http://irem.univ-reunion.fr/spip.php?article958) and click on the icon representing a paper sheet with a pencil, then elaborate further on your arguments.

Meanwhile I give here an excerpt from IEEE 754 - 2008:

> The behavior of infinity in floating-point arithmetic is derived from the limiting cases of real arithmetic with operands of arbitrarily large magnitude, when such a limit exists. Infinities shall be interpreted in the affine sense, that is: −∞ < {every finite number} < +∞.

I find it rather near the math lessons from high school !

**Comment by AlainBusser** [comment]
great :gift_heart: 
As soon as the yellow jackets allow my pupils to go to school I will show it to them
